Add notes to more accurately describe misaka library extensions

mgc8 opened the pull request:
Added note to the "math" extension to clarify that it doesn't actually render any math, just marks it as ignored by the parser, so that it can be processed with external JavaScript libraries separately. Added notes to the "highlight", "tables" and "underline" extensions with the requirements for each to actually work when used.

<!-- Just like NASA going to the moon, it's always good to have a checklist when
creating changes.
The following items are listed to help you create a great Pull Request: -->
## Checklist
- [x] All new and existing **tests are passing**
- [x] (If adding features:) I have added tests to cover my changes
- [x] (If docs changes needed:) I have updated the **documentation** accordingly.
- [ ] I have added an entry to `CHANGES.rst` because this is a user-facing change or an important bugfix
- [ ] I have written **proper commit message(s)**
      <!-- Title ideally under 50 characters (at most 72), good explanations,
      affected part of Isso mentioned in title, e.g. "docs: css: Increase font-size on buttons"
      Further info: https://github.com/joelparkerhenderson/git-commit-message -->

## What changes does this Pull Request introduce?
<!-- Explain what this PR will do, how one can try out the changes -->
Documentation changes only -- clarifications on the functionality of ``misaka`` extensions.

## Why is this necessary?
<!-- Link to existing bugs, post reproducible setups that demonstrate the
     issue/change -->
The existing documentation was incomplete.

ix5 reviewed (changes requested):
Nice, thank you!

mgc8 commented on docs/docs/reference/markdown-config.rst:
```diff
@@ -127,8 +127,14 @@ The following **additional options** are available:
+| allowing them to be processed separately with a          |                           |
+| JavaScript library.                                      |                           |
+| **Note:** This extension will *not* render equations     |                           |
+| or any form of math, it just marks them to be ignored!   |                           |
```
Fixed in latest commit.

ix5 reviewed (changes requested):
Thank you for applying the feedback.

Per `git` conventions and in order for this project to have a meaningful and well-searchable `git` history, we ask contributors to keep commit title length to 50 characters and the body to a maximum of 80.
You can look at `git log -- docs/` or browse the [history of the docs/ folder](https://github.com/isso-comments/isso/commits/master/docs) to see examples of how commits _should_ be formatted.

We also prefer you squash superfluous commits (this change should only consist of 1 logical commit since it only touches 1 file). You can do so with `git rebase -i master` and change `pick` to `squash`. Otherwise our history would be littered with multiple rounds of PR feedback, making it difficult to track down with `git blame` through which commit (and with which intentions/justification) changes to files were made.

mgc8 commented:
Sorry for the wrong format, I've made the changes locally to squash the commits and re-phrase, then pushed it over the existing PR. I hope it's fine now, please have a look and let me know if other changes are necessary.

ix5 commented:
Thank you.

One last thing, please fix the commit title length. E.g. `docs: markdown-config: Clarify misaka config` would be succinct enough.

We'd want to avoid this:
![Screenshot from 2024-04-29 23-14-31](https://github.com/isso-comments/isso/assets/10212877/6b3f37ba-e3ab-467c-8eda-dfd172be6d60)

mgc8 commented:
Thank you for taking the time on the reviews as well.

> One last thing, please fix the commit title length. E.g. `docs: markdown-config: Clarify misaka config` would be succinct enough.

Yeah, the funny thing is that I noticed that, and actually did a "reword" when I squashed the commits and in my local logs it appeared fine; but for some reason it didn't update when I pushed things to GitHub :-/ Could be that my local copy became a bit confused with all the re-basing happening from the other commits that come in, so I deleted everything, re-pulled and re-based again, I hope it's fine now?

ix5 commented:
Merged, thanks!
